services/web/server/src/simcore_service_webserver/login/_2fa.py

```diff
@@ -0,0 +1,109 @@
+""" two-factor-authentication utils
+
+Currently includes two parts:
+
+- generation and storage of secret codes for 2FA validation (using redis)
+- sending SMS of generated codes for validation (using twilio service)
+
+"""
+
+import asyncio
+import logging
+import secrets
+from typing import Optional
+
+from aiohttp import web
+from pydantic import BaseModel, Field
+from servicelib.logging_utils import log_decorator
+from twilio.rest import Client
+
+from ..redis import get_redis_validation_code_client
+from .settings import TwilioSettings
+
+log = logging.getLogger(__name__)
+
+
+class ValidationCode(BaseModel):
+    value: str = Field(..., description="The code")
+
+
+#
+# REDIS:
+#  generation, storage of secret codes
+#
+
+
+def _generage_2fa_code() -> str:
+    return f"{1000 + secrets.randbelow(8999)}"  # code between [1000, 9999)
+
+
+@log_decorator(log, level=logging.DEBUG)
+async def set_2fa_code(
+    app: web.Application,
+    user_email: str,
+    *,
+    timeout: int = 60,
```

Comment:
I think it would be useful to have this configurable via env var.

Reply:
nope ... no more timeouts in configs :-D